MIGRATION.md

```diff
@@ -0,0 +1,239 @@
+1. There is no pagination support for views. Examples coming soon.
+
+### POJO usage in the new library
+
+Since the new library supports models, this guide will provide you three ways to migrated your POJOs from `java-cloudant`.
+
+Let's start with a simple POJO:
+
+```java
+public class POJO {
+    private String _id;
+    private String name;
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    @Override
+    public String toString() {
+        return _id + ": " + name;
+    }
+}
+```
+The code block below uses the POJO in a simple read and write implementation using
+Cloudant Query to read the object and then updating the document in the database.
+
+```java
+//... set up the service client and the database
+POJO p = db.find(POJO.class, "example_id");
+System.out.println(p); // the value of the POJO's toString method
+
+p.setName("new_name");
+System.out.println(p.getName()); // will be new_name
+
+Response response = db.update(p); // the same object is used for the update, it will set the name parameter to new_name
+```
+
+#### 1. Use the `Document` model class, storing user properties in the `Map`
+
+```java
+//... set up the service client
+GetDocumentOptions documentOptions =
+                new GetDocumentOptions.Builder()
+                        .db("example_db")
+                        .docId("example_id")
+                        .build();
+
+Document doc = service.getDocument(documentOptions)
+                .execute()
+                .getResult();
+
+System.out.println(doc); // will be a JSON
+
+// Set the JSON properties to the POJO
+POJO p = new POJO();
+p.setName((String) doc.getProperties().get("name"))
+System.out.println(p); // will be a POJO with the same name as the JSON
+
+p.setName("new_name");
+System.out.println(p.getName()); // new_name
+
+doc.put("name", p.getName()); // add your modifications to the Document object
+
+PutDocumentOptions putDocumentOptions =
+        new PutDocumentOptions.Builder()
+                .db("example_db")
+                .docId("example_id")
+                .document(doc)
+                .build();
+
+DocumentResult response = service.putDocument(putDocumentOptions).execute()
+                                .getResult();
+```
+
+#### 2. Convert the `Document` model into a POJO (and vice versa)
+
+```java
+//... set up the service client
+GetDocumentOptions documentOptions =
+                new GetDocumentOptions.Builder()
+                        .db("example_db")
+                        .docId("example_id")
+                        .build();
+
+Document doc = service.getDocument(documentOptions)
+        .execute()
+        .getResult();
+
+System.out.println(doc); // will be a JSON
+
+// Serialize the JSON to POJO
+POJO p = YourJsonSerializer.fromJson(doc.toString(), POJO.class);
```

Comment:
Could we use Gson in the examples and add a note about having your own custom serializer if needed? Then users can try these examples out without needing an implementation for `YourJsonSerializer`.

Reply:
I talked about this with @ricellis offline, and he suggested using the `YourJsonSerializer`. This way we are emphasizing that the developer can use any serializer of his choice.

Reply:
Yeah in light of #352 I wanted it to be very clear that this is flexible and not restricted to GSON - I'm open to other ideas on how to do that though

Reply:
Ok makes sense.